desktop/src/com/mygdx/game/client/ecs/system/MovementSystem.java

```diff
@@ -22,23 +25,37 @@ public class MovementSystem extends IteratingSystem {
   private ComponentMapper<Highlighted> highlightedMapper;
   private ComponentMapper<Coordinates> coordinatesMapper;
   private ComponentMapper<Field> fieldMapper;
+  private final PlayerInfo playerInfo;
+  private final TurnDialogFactory turnDialogFactory;
 
   @Inject
   public MovementSystem(
-      ChosenEntity chosenEntity,
-      MoveEntityService moveEntityService
+      @NonNull ChosenEntity chosenEntity,
+      @NonNull MoveEntityService moveEntityService,
+      @NonNull PlayerInfo playerInfo,
+      @NonNull TurnDialogFactory turnDialogFactory
   ) {
     this.chosenEntity = chosenEntity;
     this.moveEntityService = moveEntityService;
+    this.playerInfo = playerInfo;
+    this.turnDialogFactory = turnDialogFactory;
   }
 
   @Override
   protected void process(int entityId) {
     if (chosenEntity.isAnyChosen() && fieldMapper.has(chosenEntity.peek())) {
       log.info("some are chosen and there's a movable highlighted entity");
       var targetCoordinate = coordinatesMapper.get(chosenEntity.pop());
-      moveEntityService.moveEntity(entityId, targetCoordinate);
+      moveEntityIfPossible(entityId, targetCoordinate);
       highlightedMapper.remove(entityId);
     }
   }
+
+  private void moveEntityIfPossible(int entityId, Coordinates targetCoordinate) {
+    if (playerInfo.isPlayerTurn()) {
+      moveEntityService.moveEntity(entityId, targetCoordinate);
+    } else {
+      turnDialogFactory.notPlayerTurnDialog();
+    }
+  }
```

Comment:
A nie lepiej blokować cały input zamiast robić to tak szczegółowo jak tutaj? W sensie, coś takiego jak tu, ale w tym systemie od inputu. Bo jakby byłby system od ataku, to on też musiałby mieć takiego checka jak tutaj, a jakby jakoś zdarzyła się sytuacja, że byłyby dwa niezależne procesy, które korzystają z inputu kliknięcia to taki dialog mógłby pojawić się dwa razy.

Reply:
hmmm nie jestem pewien, co mas z na myśli mówiąc cały input? Że nie wolna wchodzić w żadne pola nawet?

Reply:
Hmm nie wiem, to co jest teraz wydaje się ciekawsze

Reply:
zgadzam się z Kacprem, w każdej metodzie teraz trzeba będzie zwracać że to nie jest nasza tura, a można to by było odseparować już w momencie inputu, że dopóki nie mamy tokena to nic nie możemy zrobić ( poza przeglądaniem sobie mapy itp oczywiście)

Reply:
no ale to przeglądanie mapy właśnie umożliwiam, albo możemy robić wybrane akcje dzięki tej funkcjonalności, albo nie możemy robić nic

Reply:
według mnie prościej jest nie móc robić nic, ale nie wiem czy czasem przeglądanie mapy nie oznacza tylko tego, żeby móc przesuwać kamerę, a nie klikanie w submapę

Reply:
no pewnie prościej, ale nie upraszczamy wszystkiego co się da :P